prometheus/client.go

```diff
@@ -66,7 +65,28 @@ func (in *Client) GetSourceServices(namespace string, servicename string) (map[s
 	return routes, nil
 }
 
+// GetServiceMetrics returns a map of metrics (indexed by description) related to the provided service identified by its namespace and service name.
+// Returned map includes istio metrics and envoy health.
+// It returns an error on any problem.
+func (in *Client) GetServiceMetrics(namespace string, servicename string, duration string) (map[string]Metric, error) {
+	metricChan, errChan := make(chan namedMetric), make(chan error)
```

Comment:
Missing close of the channels before to leave the function ?
(not sure if perhaps it's not necessary, but just come to my attention).

Reply:
Closing channels makes code easier to read often though, since one can use for item := range metricChan { .. process items .. } just like it would have been a list. No need to return the amount of items or do the for loop against item count like later here. This is just a generic note, in this case two channels are read to it might make sense, although one could just loop over the errChan first and then quit without reading the other channel.